internal/service/advancedcluster/data_source_advanced_clusters.go

```diff
@@ -325,6 +325,11 @@ func flattenAdvancedClusters(ctx context.Context, connV220240530 *admin20240530.
 
 		zoneNameToOldReplicationSpecMeta, err := GetReplicationSpecAttributesFromOldAPI(ctx, cluster.GetGroupId(), cluster.GetName(), connV220240530.ClustersApi)
 		if err != nil {
+			if apiError, ok := admin20240530.AsError(err); !ok {
+				return nil, diag.FromErr(err)
+			} else if !(apiError.GetErrorCode() == "ASYMMETRIC_SHARD_UNSUPPORTED" && isUsingOldShardingConfiguration(d)) {
+				return nil, diag.FromErr(fmt.Errorf(errorRead, cluster.GetName(), err))
+			}
```

Comment:
in the case of the plural data source I would preserve the existing behaviour [mentioned in docs](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/data-sources/advanced_clusters#use_replication_spec_per_shard-1), if cluster is asymmetric we simply do not include in the results